src/linagora.esn.unifiedinbox/app/services/composer-status/composer-status-service.spec.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+/* global chai, sinon: false */
+
+const { expect } = chai;
+
+describe('The composerStatus service', function() {
+  let $rootScope, $log, inboxComposerStatus, INBOX_COMPOSER_STATUS;
+
+  beforeEach(function() {
+    angular.mock.module('linagora.esn.unifiedinbox');
+
+    angular.mock.inject(function(_$rootScope_, _$log_, _inboxComposerStatus_, _INBOX_COMPOSER_STATUS_) {
+      $rootScope = _$rootScope_;
+      $log = _$log_;
+      inboxComposerStatus = _inboxComposerStatus_;
+      INBOX_COMPOSER_STATUS = _INBOX_COMPOSER_STATUS_;
+
+      $rootScope.$broadcast = sinon.stub();
+      $log.error = sinon.stub();
+    });
+  });
+
+  describe('The DISCARDING status', function() {
+    it('should be INBOX_COMPOSER_STATUS.DISCARDING', function() {
+      const currentStatus = inboxComposerStatus.getStatus();
+
+      expect(currentStatus).to.equal(INBOX_COMPOSER_STATUS.DISCARDING);
+    });
+  });
+
+  describe('The getStatus method', function() {
+    it('should return the current status', function() {
+      let currentStatus = inboxComposerStatus.getStatus();
+
+      expect(currentStatus).to.equal(INBOX_COMPOSER_STATUS.DISCARDING);
+
+      inboxComposerStatus.updateStatus(INBOX_COMPOSER_STATUS.OPENING);
+
+      currentStatus = inboxComposerStatus.getStatus();
+
+      expect(currentStatus).to.equal(INBOX_COMPOSER_STATUS.OPENING);
+    });
+  });
+
+  describe('The updateStatus method', function() {
+    it('should update the current status if the incoming new status is valid and broadcast an event with the updated status', function() {
```

Comment:
Can the broadcast be tested? (same on next test cases)

Reply:
i really dont remember why there is a broadcast in the test .I updated the messages in  tests in line with the functionality